crates/rethnet_eth/src/remote/client.rs

```diff
@@ -242,9 +242,13 @@ impl RpcClient {
     where
         T: for<'a> serde::Deserialize<'a>,
     {
+        let cache_key = CacheableMethodInvocation::try_from(input)
+            .ok()
+            .and_then(|v| v.cache_key());
```

Comment:
I expect clippy to complain about this, but this can be:

```suggestion
            .and_then(CacheableMethodInvocation::cache_key);
```

Reply:
I like this pattern, but it doesn't work here unfortunately, bc `CacheableMethodInvocations::cache_key` takes `&CacheableMethodInvocations` but the value is owned.